iceoryx_utils/include/iceoryx_utils/cxx/expected.hpp

```diff
@@ -583,6 +583,28 @@ class expected<ValueType, ErrorType>
     /// @endcode
     expected& on_success(const std::function<void(expected&)>& callable) noexcept;
 
+    /// @brief  if the expected does contain a success value the given callable is called and
+    ///         a reference to the result is given as an argument to the callable
+    /// @param[in] callable callable which will be called if the expected contains a success value
+    /// @return const reference to the expected
+    /// @code
+    ///     someExpected.on_success([](int& result){
+    ///         std::cout << "we have a result : " << result << std::endl;
+    ///     })
+    /// @endcode
+    const expected& on_success(const std::function<void(ValueType&)>& callable) const noexcept;
```

Comment:
Hope that #86 will change this signature

Reply:
But to help you really with the usage of expected we need C++14. Then we can use ```on_success``` with the ```auto``` keyword like
```
someExpected.on_success([](auto & result){ ...
```